scipy/linalg/_generate_pyx.py

```diff
@@ -539,32 +540,67 @@ def generate_lapack_pxd(all_sigs):
 dims = {'work': '(*)', 'ab': '(ldab,*)', 'a': '(lda,*)', 'dl': '(*)',
         'd': '(*)', 'du': '(*)', 'ap': '(*)', 'e': '(*)', 'lld': '(*)'}
 
+xy_specialized_dims = {'x': '', 'y': ''}
+a_specialized_dims = {'a': '(*)'}
+special_cases = {'ladiv': xy_specialized_dims,
+                 'lanhf': a_specialized_dims,
+                 'lansf': a_specialized_dims,
+                 'lapy2': xy_specialized_dims,
+                 'lapy3': xy_specialized_dims}
 
 def process_fortran_name(name, funcname):
     if 'inc' in name:
         return name
-    xy_exclusions = ['ladiv', 'lapy2', 'lapy3']
-    if ('x' in name or 'y' in name) and funcname[1:] not in xy_exclusions:
-        return name + '(n)'
-    if name in dims:
-        return name + dims[name]
+    if ('x' in name or 'y' in name):
+        return name + special_cases.get(funcname[1:], dict()).get(name, '(n)')
+    name += special_cases.get(funcname[1:], dims).get(name, dims.get(name, ''))
     return name
 
+def fix_line_length(subroutine):
```

Comment:
Seems like this would be simpler if we integrated the line-wrapping into the code that generates the subroutine (`fort_subroutine_wrapper`). It would at least remove the need to do fiddly parsing here.

Reply:
The line wrapping is needed to get around the maximum column limitation in Fortran 77. Including it in the original code generation is a good idea though.

Reply:
Okay, so after looking at this a bit more, it turns out that the xblas functions were the only things hitting this corner case, so we could just pull this logic out entirely for LAPACK 3.4.0. That said, I don't really want to deal with this breaking again when doing some future update, so I'm pushing a version that handles the max line length issue by just vertically stacking the arguments in the generated Fortran subroutines. That greatly simplifies the logic for generating the wrapper but should also avoid potential trouble with line length later.